index.rst

```diff
@@ -42,20 +42,151 @@
+
+Metadata Driven Processing
+==========================
+
+In this scenario the data processing is completely decoupled from the observing script and decisions on the processing within the OCPS are entirely mediated by metadata associated with the observation itself.
+The control flow in the observing script could look something like this:
+
+* Script sends multiple "take images" commands in loop.
+   * This command includes specification of a groupid and pipeline name.
+   * If this is for a co-addition pipeline, for the final observation in group also include a flag indicating that data taking in this group is complete.
+
+and the OCPS will asynchronously process the data that arrives:
+
+* The Archiver/CCS issues a "data available" event with the relevant obsid once the data are available in a butler repository.
+* OCSP triggers on this "data available" event.
+* OCPS retrieves relevant metadata from the butler using this obsid.
+* Determine the pipeline name from the metadata.
+* Check whether this is from a previously known groupid or a new groupid.
+* If this is a new groupid ensure that any previous co-addition pipelines have been flushed.
+* Create quantum graph for this pipeline and send for execution.
+* If the "group complete" flag is set in the metadata, trigger the co-addition pipeline.
+
+The script has the option of changing the groupid in order to control how data are grouped in the OCPS.
+A standard mechanism will be provided to scripts for generating a new unique groupid from an existing groupid.
+A script should not modify the supplied for standard visits or alternate standard visits since the Alert Production system uses the groupid issued by the ScriptQueue as a rendezvous mechanism.
+
+If the ScriptQueue is told to stop the observing script early such that the "group complete" flag will never turn up, the ScriptQueue can send a command to the OCPS to trigger the co-addition pipeline.
+The general approach though is that the observing script does not know anything about the OCPS but can control which observations are co-added and which pipeline to use.
+This metadata-mediated processing approach is similar to that used by ORAC-DR :cite:`2015A&C.....9...40J`.
+
+Using Processed Data
+====================
+
+Once data have been processed by the OCPS an event should be issued when each dataset type becomes available.
+The event could trigger the observatory visualization system allowing specific dataset types and detectors to be selected for immediate display as part of the summit quality control process.
+The observatory visualization system configuration can therefore be loosely coupled to the OCPS and how to display a particular dataset type is entirely configuration driven within the visualization system.
+
+Notebooks can also block waiting for the processed data before doing bespoke commissioning visualization or further detailed analysis that is decoupled from observing.
```

Comment:
These notebooks would have to share Butler datastores and registries with the OCPS.  With the possibility of multiple processing platforms being available (Camera Diagnostic Cluster, Commissioning Cluster, NCSA Prompt enclave), we need to be careful that these are properly associated.